src/language-js/comments.js

```diff
@@ -22,59 +22,59 @@ const {
 } = require("./utils");
 const { locStart, locEnd } = require("./loc");
 
-function handleOwnLineComment(comment, text, options, ast, isLastComment) {
+function handleOwnLineComment(context, text, options, ast, isLastComment) {
```

Comment:
Maybe let's move everything to one argument: `{ comment, precedingNode, enclosingNode, followingNode, text, options, ast, isLastComment }`?

Reply:
I thought about this too, 

> Rename `data` -> `context` …
> I'm thinking maybe we want add a new `addLeadingComment` function through `context`

https://github.com/prettier/prettier/pull/9726/commits/cf97d08e76a7bb74e19079399cb84ca7ff153674

Reply:
I think `add*Comment` and `is*Comment` should be WeakMap-based. No need to pass them around then.

Reply:
Yes, maybe